axiom/vfields_integration_test.go

```diff
@@ -0,0 +1,121 @@
+package axiom_test
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/suite"
+
+	"github.com/axiomhq/axiom-go/axiom"
+)
+
+// VirtualFieldsTestSuite tests all methods of the Axiom Virtual Fields API
+// against a live deployment.
+type VirtualFieldsTestSuite struct {
+	IntegrationTestSuite
+
+	// Setup once per test.
+	vfield *axiom.VirtualFieldWithID
+}
+
+func TestVirtualFieldsTestSuite(t *testing.T) {
+	suite.Run(t, new(VirtualFieldsTestSuite))
+}
+
+func (s *VirtualFieldsTestSuite) SetupSuite() {
+	s.IntegrationTestSuite.SetupSuite()
+}
+
+func (s *VirtualFieldsTestSuite) TearDownSuite() {
+	s.IntegrationTestSuite.TearDownSuite()
+}
+
+func (s *VirtualFieldsTestSuite) SetupTest() {
+	s.IntegrationTestSuite.SetupTest()
+
+	var err error
+	s.vfield, err = s.client.VirtualFields.Create(s.ctx, axiom.VirtualField{
+		Dataset:    "test-dataset",
```

Comment:
I assume the dataset doesn't actually need to exist?

From a users standpoint, not sure if that is a nice API 😅

Reply:
It will return error if dataset doesn't exist

Reply:
But in that case, this test will error, no? Where is `test-dataset` being created.